### DISCUSSION TITLE
How do I rename a bluetooth device I already enrolled

### DISCUSSION THREAD

**Discussion by ajbastien** [ESPresense/ESPresense#1177] [Support]
I enrolled 2 iPhones in ESPresense.  The first one went fine.  The second one was added but shows up with a blank Alias in the devices page.  How do I correct this?  I tried enrolling the same device again but it is still blank.
Alias and Name column are blank for that device.

I did try publishing to the espresense/settings/irk:****/config topic with the correction.  It did work but after the device restarted it went back to blank again.

**Comment by supaeasy** [comment]
Your publishing approach was right. But make sure to retain the message. Then it will be consistent.

**Comment by ripvega** [comment]
This is just what I need to do.

Could someone please help me out with a detailed step-by-step of what to do?

**Comment by ajbastien** [comment]
Download https://mqtt-explorer.com/
Publish a message to the topic: espresense/settings/irk:hhhhh/config (where irk:hhhhh is your device IRK for the iPhone/watch)
Pick json as the payload
Check the retain flag

Payload example (enter the id and name you want):
{
  "id": "myphone",
  "name": "MyPhone"
}
![image](https://github.com/ESPresense/ESPresense/assets/8333336/60295ea2-2b9a-4903-9cb9-7c0991348386)

**Reply by ripvega** [reply]
Thank you so much! That worked a treat. I appreciate your time.

For future reference, where can I look to find out how to delete a device?

**Reply by supaeasy** [reply]
Delete given devices Topic in MQTT Explorer. 
You can play around with it a bit to get a feeling how this works. There is nothing you can really break.
